SeaFightLogic/Ships/Ship.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SeaFight
+{
+
+    public abstract class Ship
+    {
+        public int Speed { get; protected set; }
+        public int ActionDistance { get; protected set; }
+        public int Size { get; protected set; }
+
+        public Ship(int size, int speed, int actionDistance)
+        {
+            if (size < 1 || size > 5)
+            {
+                throw new ArgumentOutOfRangeException("Size of ship must be from 1 to 5");
+            }
+            if (speed < 1 || speed > 5)
+            {
+                throw new ArgumentOutOfRangeException("Speed of ship must be from 1 to 5");
+            }
+            if (actionDistance < 1 || actionDistance > 5)
+            {
+                throw new ArgumentOutOfRangeException("Size of ship must be from 1 to 5");
+            }
+
+            Speed = speed;

```

Comment:
Always write "this" if the property belongs to this class. Microsoft recommendation

Reply:
done